ksqldb-engine/src/main/java/io/confluent/ksql/services/DefaultConnectClientFactory.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.services;
+
+import io.confluent.ksql.services.ConnectClient.ConnectClientFactory;
+import io.confluent.ksql.util.FileWatcher;
+import io.confluent.ksql.util.KsqlConfig;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Paths;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import org.apache.kafka.common.config.ConfigException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultConnectClientFactory implements ConnectClientFactory {
```

Comment:
State changes for `connectAuthHeader` a bit non-trivial, so I would probably add a short javadoc describing it. Something along the lines:
* `connectAuthHeader` is set on the first `get`.
* If credentials are in a file, we read them from the file AND start a file watcher thread that monitors file changes and updates the header accordingly.